packages/compiler/src/collector/expression.ts

```diff
@@ -1,49 +1,124 @@
 import { Node, ts } from "ts-morph";
 import { match } from "ts-pattern";
+import * as types from "../types";
 
-export const handleJsxExpression = (node: Node<ts.Node>) => {
+/**
+ * Normalize ts-morph Node to make further processing simpler
+ */
+const normalizeNode = (node: Node): Node => {
```

Comment:
Note: I copied this from `decode.ts`, which is now removed. Now this is used internally by `handleJsxExpression` consistently. Previous pattern was error prone since people might forget to call `decode` where necessary. I think there is no reason to split the call between `decode` and `normalizeNode`. Note that in the future probably we can replace this kind of logic with some library such as ts-evaluate